pkg/server/middleware/request_limits.go

```diff
@@ -22,11 +22,11 @@ func (l RequestLimits) Wrap(next http.Handler) http.Handler {
 		reader := io.LimitReader(r.Body, int64(l.maxRequestBodySize)+1)
 		body, err := io.ReadAll(reader)
 		if err != nil {
-			http.Error(w, fmt.Sprintf("failed to read request body: %v", err), http.StatusInternalServerError)
+			http.Error(w, fmt.Sprintf("failed to read request body: %v", err), http.StatusBadRequest)
```

Comment:
just considering this - this `err` is actually `500` - we could not read the request body (we don't know why).

I think this is separate from the case where the body size is greater than the amount we accept (we've read the request, we see it's 1 byte longer than our max, so we stopped reading is the case below).

Reply:
returned 500 back, thank you!